tests/Integration/inc/classes/preload/Sitemap/runPreload.php

```diff
@@ -0,0 +1,244 @@
+	protected $homeWpOption;
+	protected $preloadErrorsTransient;
+	protected $preloadRunningTransient;
+	protected $process;
+
+	public static function setUpBeforeClass() {
+		require_once WP_ROCKET_TESTS_FIXTURES_DIR . '/Preload/Process_Wrapper.php';
+	}
+
+	public function setUp() {
+		parent::setUp();
+
+		$this->homeWpOption            = get_option( 'home' );
+		$this->preloadErrorsTransient  = get_transient( 'rocket_preload_errors' );
+		$this->preloadRunningTransient = get_transient( 'rocket_preload_running' );
+		$this->process                 = new Process_Wrapper();
+
+		update_option( 'home', 'https://example.com/' );
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		if ( $this->process ) {
+			// Added by \WP_Async_Request.
+			remove_action( 'wp_ajax_' . $this->identifier, [ $this->process, 'maybe_handle' ] );
+			remove_action( 'wp_ajax_nopriv_' . $this->identifier, [ $this->process, 'maybe_handle' ] );
+			// Added by \WP_Background_Process.
+			remove_action( $this->identifier . '_cron', [ $this->process, 'handle_cron_healthcheck' ] );
+			remove_filter( 'cron_schedules', [ $this->process, 'schedule_cron_healthcheck' ] );
+		}
+
+		foreach ( [ 'manual_preload', 'cache_mobile', 'do_caching_mobile_files' ] as $option ) {
+			remove_filter( $this->option_hook_prefix . $option, [ $this, 'return_0' ] );
+			remove_filter( $this->option_hook_prefix . $option, [ $this, 'return_1' ] );
+		}
+
+		remove_filter( $this->option_hook_prefix . 'cache_reject_uri', [ $this, 'return_empty_array' ] );
+		remove_filter( $this->option_hook_prefix . 'cache_query_strings', [ $this, 'return_cache_query_strings' ] );
+
+		update_option( 'home', $this->homeWpOption );
+		set_transient( 'rocket_preload_errors', $this->preloadErrorsTransient );
+		set_transient( 'rocket_preload_running', $this->preloadRunningTransient );
+
+		$this->homeWpOption            = null;
+		$this->preloadErrorsTransient  = null;
+		$this->preloadRunningTransient = null;
+		$this->process                 = null;
+	}
+
+	public function testShouldNotPreloadWhenNoUrls() {
+		$preload_process = $this->createMock( Full_Process::class );
+
+		Actions\expectDone( 'before_run_rocket_sitemap_preload' )->never();
+
+		// No URLs.
+		( new Sitemap( $preload_process ) )->run_preload( [] );
+	}
+
+	public function testShouldPreloadSitemapsWhenValidUrls() {
+		$sitemaps = [
+			'https://example.com/sitemap.xml',
+			'https://example.com/sitemap-mobile.xml',
+		];
+
+		// Fake the requests.
+		Functions\when( 'wp_remote_get' )->alias( function( $url, $args = [] ) {
+			$path = WP_ROCKET_TESTS_FIXTURES_DIR . '/Preload/Sitemap/';
+
+			switch ( $url ) {
+				case 'https://example.com/sitemap.xml':
+					$path .= 'sitemap.xml';
+					break;
+				case 'https://example.com/sitemap-mobile.xml':
+					$path .= 'sitemap-mobile.xml';
+					break;
+				default:
+					return new \WP_Error( 'wrong-url', 'Wrong URL', [ $url ] );
+			}
+
+			if ( ! \file_exists( $path ) ) {
```

Comment:
No need for the `\` as `file_exists()` is a PHP native function that is in global space. PHP will know how to find it.